### PR TITLE
Update prance library usage

### PR DESCRIPTION
**Pull request by ahitacat**
# Overview

This PR is being created to update the usage of the prance library. The translating parser is already part of Prance upstream, but the Inventory did not use it. Instead, was having its own copy of the class. 

This PR removes the translating_parser library which contained the code that is already in Prance upstream.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_unit.py [hunk 1]
```diff
@@ -443,6 +444,24 @@ def test_specification_is_parsed(self, get_engine, app):
...
         assert args[0] is not None
 
+    # TODO: test here the parsing is working with the $defs from system_profile.spec.yaml
+    # and the check parser.specification["components"]["schemas"] - this is more a library test
+    @patch("app.create_app")
+    def test_translatingparser(self, translating_parser, get_engine, app):
+        create_app(RuntimeEnvironment.TEST)
+        # check if SystemProfile is in the componentes referenced
+        args = app.return_value.add_api.mock_calls[0].args
+        assert "SystemProfile" in args[0]["components"]["schemas"]
```
Comment (Glutexo) [inline tests/test_unit.py:454]: Testing _SystemProfile_ schema does not really test that the files are compiled together and that the translation happened. This schema is already present in the [_api.spec.yaml_](https://github.com/ahitacat/insights-host-inventory/blob/47de59e131c2b0645a0f0ee812ab94d60c3032db/swagger/api.spec.yaml#L1089) root document. What we need to test or is an object that gets copied from [system_profile.spec.yaml](https://github.com/ahitacat/insights-host-inventory/blob/update-prance-referencies/swagger/system_profile.spec.yaml), e.g. [_DiskDevice_](https://github.com/ahitacat/insights-host-inventory/blob/47de59e131c2b0645a0f0ee812ab94d60c3032db/swagger/system_profile.spec.yaml#L17). If the translation worked, it would be copied under the name _system_profile.spec.yaml_DiskDevice_.

Now, in [_openapi.json_](https://github.com/ahitacat/insights-host-inventory/blob/47de59e131c2b0645a0f0ee812ab94d60c3032db/swagger/openapi.json) it is under [_SystemProfileDiskDevice_](https://github.com/ahitacat/insights-host-inventory/blob/47de59e131c2b0645a0f0ee812ab94d60c3032db/swagger/openapi.json#L2178). I’d test for that and add a comment.

**Review by Glutexo — CHANGES_REQUESTED** [review]

File: tests/test_unit.py (hunk 1 above)
Comment (ahitacat) [reply tests/test_unit.py:454]: Oh, yes you are right. I'm going to modify to check the nested ones

File: tests/test_unit.py [hunk 2]
```diff
@@ -443,6 +444,24 @@ def test_specification_is_parsed(self, get_engine, app):
         assert len(args) == 1
         assert args[0] is not None
 
+    # TODO: test here the parsing is working with the $defs from system_profile.spec.yaml
+    # and the check parser.specification["components"]["schemas"] - this is more a library test
+    @patch("app.create_app")
```
[the remark this answers is not in the document]
Comment (ahitacat) [reply tests/test_unit.py:449]: Now you ask, I am not quite sure. I am trying to get what is inside of parser.specifications, which is inside of `create_app.` Do you mean that this test is using the `connexion.App `and it should be taking the result of the inside mock of `parser.specifications`?

[the remark this answers is not in the document]
Comment (Glutexo) [reply tests/test_unit.py]: You have the specification in `app.return_value.add_api.mock_calls[0].args[0]`. The `create_app` call still uses the real function and it should be like that. It is this function that is being tested, that means it doesn’t make sense to mock it. Moreover the patch does not work, because it mocks only cases when the `create_app ` name is used in the `app` module. In this case we import `create_app` here in the tests before the patching happens.

[the remark this answers is not in the document]
Comment (ahitacat) [reply tests/test_unit.py]: I think the new commit address this well.

**Review by thearifismail — APPROVED** [review]

**Comment by ahitacat** [comment]
> I resolved the conflicts and fixed some dependency issues. @ahitacat Please take a look to see if I messed anything up!

I don't see anything messed up :smile:

**Review by kruai — APPROVED** [review]
